test/Common/DurableTaskEndToEndTests.cs

```diff
@@ -2659,6 +2659,95 @@ public async Task TaskHubName_Throws_ArgumentException(string taskHubName)
                 argumentException.Message);
         }
 
+        [Theory]
+        [Trait("Category", PlatformSpecificHelpers.TestCategory)]
+        [InlineData(true)]
+        [InlineData(false)]
+        public async Task Dedupe_NotRunning_ThrowsException(bool extendedSessions)
+        {
+            var instanceId = "OverridableStatesThrowsExceptionNotRunningTest";
+
+            using (ITestHost host = TestHelpers.GetJobHost(
+                 this.loggerProvider,
+                 nameof(this.Dedupe_NotRunning_ThrowsException),
+                 extendedSessions,
+                 overridableStates: OverridableStates.NonRunningStates))
+            {
+                await host.StartAsync();
+
+                int initialValue = 0;
+
+                var client = await host.StartOrchestratorAsync(nameof(TestOrchestrations.Counter), initialValue, this.output, instanceId: instanceId);
+
+                // Wait for the instance to go into the Running state. This is necessary to ensure log validation consistency.
+                await client.WaitForStartupAsync(this.output);
+
+                TimeSpan waitTimeout = TimeSpan.FromSeconds(Debugger.IsAttached ? 300 : 10);
+
+                // Perform some operations
+                await client.RaiseEventAsync("operation", "incr", this.output);
+                await client.WaitForCustomStatusAsync(waitTimeout, this.output, 1);
+
+                // Make sure it's still running and didn't complete early (or fail).
+                var status = await client.GetStatusAsync();
+                Assert.True(
+                    status?.RuntimeStatus == OrchestrationRuntimeStatus.Running ||
+                    status?.RuntimeStatus == OrchestrationRuntimeStatus.ContinuedAsNew);

```

Comment:
The rest of the test assumes that the status here is running, not continued as new, so I'm not sure if we need the second half of this predicate.